client/cli/src/params/shared_params.rs

```diff
@@ -93,20 +88,26 @@ impl SharedParams {
 	/// 1. Set the panic handler
 	/// 2. Raise the FD limit
 	/// 3. Initialize the logger
-	pub fn init(&self, version: &VersionInfo) -> error::Result<()> {
-		crate::init(self.log.as_ref().map(|v| v.as_ref()).unwrap_or(""), version)
+	pub fn init<C: SubstrateCli>(&self) -> Result<()> {
+		let logger_pattern = self.log.as_ref().map(|v| v.as_ref()).unwrap_or("");
+
+		sp_panic_handler::set(C::support_url(), C::impl_version());
+
+		fdlimit::raise_fd_limit();
+		init_logger(logger_pattern);
+
+		Ok(())
 	}
-}
 
-fn base_path(cli: &SharedParams, version: &VersionInfo) -> PathBuf {
-	cli.base_path.clone()
-		.unwrap_or_else(||
-			app_dirs::get_app_root(
-				AppDataType::UserData,
-				&AppInfo {
-					name: version.executable_name,
-					author: version.author
-				}
-			).expect("app directories exist on all supported platforms; qed")
-		)
+	/// Get the database configuration object for the parameters provided
+	pub fn database_config(
+		&self,
+		base_path: &PathBuf,
+		cache_size: usize,
+	) -> Result<DatabaseConfig> {
```

Comment:
Why does this returns a result?

Reply:
CliConfiguration is meant to be fallible for any field. This allows the binary to customize the behavior during the parsing and decide to fail if some conditions are met. It's just to be generic.